client/api.go

```diff
@@ -1050,7 +1051,277 @@ func (c *Client) TestSendAnonVotes(
+	var req api.APIrequest
+	var err error
+	req.Method = "submitRawTx"
+
+	tx := &models.SetAccountInfoTx{
+		Txtype:  models.TxType_SET_ACCOUNT_INFO,
+		Nonce:   nonce,
+		InfoURI: infoURI,
+		Account: to.Bytes(),
+	}
+
+	stx := &models.SignedTx{}
+	stx.Tx, err = proto.Marshal(&models.Tx{Payload: &models.Tx_SetAccountInfo{SetAccountInfo: tx}})
+	if err != nil {
+		return err
+	}
+	if stx.Signature, err = signer.SignVocdoniTx(stx.Tx); err != nil {
+		return err
+	}
+	if req.Payload, err = proto.Marshal(stx); err != nil {
+		return err
+	}
+
+	resp, err := c.Request(req, nil)
+	if err != nil {
+		return err
+	}
+	if !resp.Ok {
+		return fmt.Errorf("%s failed: %s", req.Method, resp.Message)
+	}
+	return nil
+}
+
+func (c *Client) GetAccount(signer *ethereum.SignKeys, accountAddr common.Address) (*vochain.Account, error) {
+	req := api.APIrequest{Method: "getAccount", EntityId: accountAddr.Bytes()}
+	resp, err := c.Request(req, signer)
+	acc := &vochain.Account{}
+	if err != nil {
+		return nil, err
+	}
+	if !resp.Ok {
+		return nil, fmt.Errorf("could not get account: %s", resp.Message)
+	}
+	if resp.Balance != nil {
+		acc.Balance = *resp.Balance
+	}
+	if resp.Nonce != nil {
+		acc.Nonce = *resp.Nonce
+	}
+	if resp.InfoURI != "" {
+		acc.InfoURI = resp.InfoURI
+	}
+	return acc, nil
+}
+
+func (c *Client) GetTreasurer(signer *ethereum.SignKeys) (*models.Treasurer, error) {
+	req := api.APIrequest{Method: "getTreasurer"}
+	resp, err := c.Request(req, signer)
+	treasurer := &models.Treasurer{}
+	if err != nil {
+		return nil, err
+	}
+	if !resp.Ok {
+		return nil, fmt.Errorf("could not get treasurer: %s", resp.Message)
+	}
+	if resp.EntityID != "" {
+		treasurer.Address = common.HexToAddress(resp.EntityID).Bytes()
+	}
+	if resp.Nonce != nil {
+		treasurer.Nonce = *resp.Nonce
+	}
+	return treasurer, nil
+}
+
+// del = true -> delete operation
+func (c *Client) SetAccountDelegate(signer *ethereum.SignKeys, delegate common.Address, nonce uint32, del bool) error {
+	var req api.APIrequest
+	var err error
+	req.Method = "submitRawTx"
+
+	tx := &models.SetAccountDelegateTx{
+		Txtype:   models.TxType_ADD_DELEGATE_FOR_ACCOUNT,
+		Nonce:    nonce,
+		Delegate: delegate.Bytes(),
+	}
+
+	if del {
+		tx.Txtype = models.TxType_DEL_DELEGATE_FOR_ACCOUNT
+	}
+
+	stx := &models.SignedTx{}
+	stx.Tx, err = proto.Marshal(&models.Tx{Payload: &models.Tx_SetAccountDelegateTx{SetAccountDelegateTx: tx}})
+	if err != nil {
+		return err
+	}
+	if stx.Signature, err = signer.SignVocdoniTx(stx.Tx); err != nil {
+		return err
+	}
+	if req.Payload, err = proto.Marshal(stx); err != nil {
+		return err
+	}
+
+	resp, err := c.Request(req, nil)
+	if err != nil {
+		return err
+	}
+	if !resp.Ok {
+		return fmt.Errorf("%s failed: %s", req.Method, resp.Message)
+	}
+	return nil
+}
+
+func (c *Client) SendTokens(from *ethereum.SignKeys, to common.Address, value uint64, nonce uint32) error {
+	var req api.APIrequest
+	var err error
+	req.Method = "submitRawTx"
+
```

Comment:
There is some repeated code here. You could move most of the code to a new method such as `c.SignAndSendTx(tx models.Tx)`